src/Security/Authentication/OpenIdConnect/samples/OpenIdConnectSample/Startup.cs

```diff
@@ -266,7 +266,7 @@ await context.SignOutAsync(OpenIdConnectDefaults.AuthenticationScheme, new Authe
                     // Persist the new acess token
                     props.UpdateTokenValue("access_token", payload.RootElement.GetString("access_token"));
                     props.UpdateTokenValue("refresh_token", payload.RootElement.GetString("refresh_token"));
-                    if (payload.RootElement.TryGetProperty("expires_in", out var property) && property.TryGetInt32(out var seconds))
+                    if (payload.RootElement.TryGetProperty("expires_in", out var property) && int.TryParse(property.GetString(), out var seconds))
```

Comment:
Why this change?

Reply:
property.TryGetInt32 was throwing a InvalidOperationException on ValueKind not being a number

Reply:
https://github.com/dotnet/runtime/blob/0f56e166b16100c23dc81ae082f6155362b7c596/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs#L481

is the check that ensures it is a Number.

This seems like a bug. There should be no reason for a `TryGet` method to throw an exception.